pkg/ui/aim/embed/custom.patch

```diff
@@ -638,3 +774,105 @@ index 0fc79c5..867376b 100644
+   EXPERIMENTS: 'experiments',
+   GET_EXPERIMENT_BY_ID: (id: string) => `experiments/${id}`,
+   UPDATE_EXPERIMENT_BY_ID: (id: string) => `experiments/${id}`,
++  DELETE_EXPERIMENT: (id: string) => `experiments/${id}`,
+   SEARCH_EXPERIMENT: (query: string) => `experiments/search=${query}`,
+   GET_RUNS_BY_EXPERIMENT_ID: (id: string) => `experiments/${id}/runs`,
+ };
+@@ -44,6 +45,9 @@ function getRunsOfExperiment(
+   return API.get(endpoints.GET_RUNS_BY_EXPERIMENT_ID(id), params);
+ }
+ 
++function deleteExperiment(id: string) {
++  return API.delete(endpoints.DELETE_EXPERIMENT(id));
++}
+ const experimentsService = {
+   endpoints,
+   getExperimentsData,
+@@ -52,6 +56,7 @@ const experimentsService = {
+   updateExperimentById,
+   createExperiment,
+   getRunsOfExperiment,
++  deleteExperiment,
+ };
+ 
+ export default experimentsService;
+diff --git a/aim/web/ui/src/services/models/experiments/experimentsAppModel.ts b/aim/web/ui/src/services/models/experiments/experimentsAppModel.ts
+new file mode 100644
+index 0000000..250bb04
+--- /dev/null
++++ b/aim/web/ui/src/services/models/experiments/experimentsAppModel.ts
+@@ -0,0 +1,66 @@
++import _ from 'lodash-es';
++
++import experimentsService from 'services/api/experiments/experimentsService';
++
++import { INotification } from 'types/components/NotificationContainer/NotificationContainer';
++
++import exceptionHandler from 'utils/app/exceptionHandler';
++
++import createModel from '../model';
++
++const model = createModel<Partial<any>>({});
++
++function initialize() {
++  model.init();
++}
++function deleteExperiment(id: string, successCallback: () => void = _.noop) {
++  try {
++    experimentsService
++      .deleteExperiment(id)
++      .call((detail) => {
++        exceptionHandler({ model, detail });
++      })
++      .then((res: any) => {
++        if (res.id) {
++          successCallback();
++        } else {
++          onNotificationAdd({
++            id: Date.now(),
++            severity: 'error',
++            messages: ['Something went wrong'],
++          });
++        }
++      });
++  } catch (err: any) {
++    onNotificationAdd({
++      id: Date.now(),
++      severity: 'error',
++      messages: [err.message],
++    });
++  }
++}
++
++function onNotificationDelete(id: number) {
++  let notifyData: INotification[] | [] = model.getState()?.notifyData || [];
++  notifyData = [...notifyData].filter((i) => i.id !== id);
++  model.setState({ notifyData });
++}
++
++function onNotificationAdd(notification: INotification) {
++  let notifyData: INotification[] | [] = model.getState()?.notifyData || [];
++  notifyData = [...notifyData, notification];
++  model.setState({ notifyData });
++  setTimeout(() => {
++    onNotificationDelete(notification.id);
++  }, 3000);
++}
++
++const runDetailAppModel = {
++  ...model,
++  initialize,
++  deleteExperiment,
++  onNotificationAdd,
++  onNotificationDelete,
++};
++
++export default runDetailAppModel;
```

Comment:
should this not belong to `aim/web/ui/src/pages/Experiment/ExperimentStore.ts` instead?

Reply:
I was following the approach they use with run deletion in `aim/web/ui/src/pages/RunDetail/RunDetailSettingsTab/RunDetailSettingsTab.tsx`